### DISCUSSION TITLE
We can't delete the cluster

### DISCUSSION THREAD

**Discussion by clickersmudge** [kubermatic/kubermatic#11783] [Q&A] [answered]
Hi,
We wanted to delete one of the test clusters. 
Unfortunately, the cluster is not deleted.
![obraz](https://user-images.githubusercontent.com/113194014/215162801-f8d018bf-ae8d-4b72-bdb2-3c09cfd54956.png)
What could be the cause? How to remove the cluster along with the resources.

**Comment by kron4eg** [comment, marked as answer]
@clickersmudge you need to check `kubectl describe cluster <YOUR CLUSTER ID>`, most likely it has some finalizers that can not be completed (deleted api key for example) and don't let Cluster object to be deleted.
